ash/src/entry.rs

```diff
@@ -42,7 +41,7 @@ pub struct EntryCustom<L> {
 
 #[derive(Debug)]
 pub enum LoadingError {
-    LibraryLoadError(io::Error),
+    LibraryLoadError(libloading::Error),
```

Comment:
I am not so sure about that change. I don't really want to leak types of other crates into the public interface of ash. 
- This is a breaking change
- Couplse ash and libloading. Updating libloading might also be a breaking change for ash in the future.

I am not too sure what the best practice here would be.

Reply:
I think the idiomatic solution is to wrap `libloading::Error` in an ash-defined newtype that implements `Display` and `Error`. Or, since this is an enum with a single case, just replace the enum with such a newtype directly.

Reply:
I think we can tackle this problem elsewhere. Currently `new_custom` looks this this
```rust
pub fn new_custom<Open, Load>(open: Open, mut load: Load) -> Result<Self, LoadingError>
    where
        Open: FnOnce() -> Result<L, LoadingError>,
        Load: FnMut(&mut L, &::std::ffi::CStr) -> *const c_void { // ... }
```
which is kind of nonsensical because we explicitly move the error handling *into* the function. I think it should look like this:
```rust
pub fn new_custom<Load>(lib: L, mut load: Load) -> Self
    where
        Load: FnMut(&mut L, &::std::ffi::CStr) -> *const c_void { // ... }
```
This way the user can handle the error of creating the lib in his own code. Then we could change the signature of the `libloading` based `Entry` new function to this
```rust
pub fn new() -> std::result::Result<CoreLoader<libloading::Library>, libloading::Error> { // ... }
```
Now we only expose `libloading` types in `#[cfg(feature = "loading")]` code

Reply:
That's a separate issue. We still need to newtype the error to avoid gratuitous breakage.

Reply:
Yes, my proposal would still be breaking, but wouldn't it be better if we fixed the root of the issue instead of working around it?

Reply:
The root of the issue is that a foreign crate's type, namely `libloading::Error`, is being added to ash's public interface. Wrapping a newtype around it is the necessary fix.

Reply:
@Ralith I implemented your suggested change in 36a3942f0426f0aee21d0d9a74470bdcca06a157